src/site/stages/build/drupal/api.js

```diff
@@ -1,23 +1,44 @@
 const moment = require('moment');
-
 const fetch = require('node-fetch');
 
 const GET_ALL_PAGES = require('./graphql/GetAllPages.graphql');
 const GET_PAGE_BY_ID = require('./graphql/GetPageById.graphql');
 const GET_LATEST_PAGE_BY_ID = require('./graphql/GetLatestPageById.graphql');
 
+const ENVIRONMENTS = require('../../../constants/environments');
 const DRUPALS = require('../../../constants/drupals');
 
-function encodeCredentials({ username, password }) {
-  const credentials = `${username}:${password}`;
+function encodeCredentials({ user, password }) {
+  const credentials = `${user}:${password}`;
   const credentialsEncoded = Buffer.from(credentials).toString('base64');
   return credentialsEncoded;
 }
 
 function getDrupalClient(buildOptions) {
-  const { address, credentials } = DRUPALS[buildOptions.buildtype];
+  const buildArgs = {
+    address: buildOptions['drupal-address'],
+    user: buildOptions['drupal-user'],
+    password: buildOptions['drupal-password'],
+  };
+
+  Object.keys(buildArgs).forEach(key => {
+    if (!buildArgs[key]) delete buildArgs[key];
+  });
+
+  const envConfig = DRUPALS[buildOptions.buildtype];
+  const drupalConfig = {};
+
+  if (buildOptions.buildtype === ENVIRONMENTS.LOCALHOST) {
+    // On localhost, build args can override the default environment.
+    Object.assign(drupalConfig, envConfig, buildArgs);
+  } else {
+    // Otherwise, properties from the hardcoded env configs should take priority.
```

Comment:
Do we need to have this difference? Presumably we wouldn't set the build args in the Jenkinsfile, so env variables would be used.

Reply:
`envConfig` is kinda misleading, because it's actually referring to the hardcoded build-type/environment setup we do in `drupals.js`, instead of actual environment variables. The actual environment variables are just default value for the command args.

Reply:
I'll try to clarify around this. The condition is only necessary so that localhost can use build args but fall back to Drupal Staging in `drupal.js` otherwise.

Reply:
Actually, I think I'm going to leave this as-is, because on localhost you can set an environment variable `DRUPAL_ADDRESS` set to Drupal Dev, and borrow the user/password from the localhost default, Drupal Staging.